neo4j/src/main/java/org/apache/metamodel/neo4j/Neo4jDataSet.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.metamodel.neo4j;
+
+import org.apache.metamodel.data.AbstractDataSet;
+import org.apache.metamodel.data.DefaultRow;
+import org.apache.metamodel.data.Row;
+import org.apache.metamodel.data.SimpleDataSetHeader;
+import org.apache.metamodel.query.SelectItem;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+final class Neo4jDataSet extends AbstractDataSet {
+
+    private JSONObject _resultJSONObject;
+    private int _currentRowIndex;
+    private Row _row;
+
+    public Neo4jDataSet(SelectItem[] selectItems, JSONObject resultJSONObject) {
```

Comment:
That depends a bit on what it actually means right now. Does it work on large datasets at all? I would expect there to be some paging implemented by default (say, more than 1000 or 10000 rows or so) which means that this design is defect and will cut off a part of the dataset. That's such a big issue/bug that I would vote against it. But if it is functional right now and it's "just" a memory issue then I could live with it for now under the saying "first make it functional, then make it optimal".